sdk/go/common/resource/plugin/host.go

```diff
@@ -164,6 +164,13 @@ func NewDefaultHost(ctx *Context, runtimeOptions map[string]interface{},
 }
 
 func parsePluginOpts(providerOpts workspace.PluginOptions, k workspace.PluginKind) (workspace.ProjectPlugin, error) {
+	handleErr := func(msg string, a ...interface{}) (workspace.ProjectPlugin, error) {
+		return workspace.ProjectPlugin{},
+			fmt.Errorf("parsing plugin options for '%s': %w", providerOpts.Name, fmt.Errorf(msg, a...))
+	}
```

Comment:
P.S. I saw the review after the PR merged. 

I view this as more of a style choice. Named return values have some confusing semantics with empty returns, so I avoid them in the general case. I also like the ability to avoid specifying `workspace.ProjectPlugin{}` at each exit point.

Reply:
> P.S. I saw the review after the PR merged.

Not a problem! I realized the merge command was already in by the time I submitted my comments.

> Named return values have some confusing semantics with empty returns, so I avoid them in the general case

I'm curious for more details on that when you have a chance.
I've found modifying error returns from a `defer` statement to be a common pattern
so I'm interested in following up on any issues/confusion around it.

> I also like the ability to avoid specifying workspace.ProjectPlugin{} at each exit point.

Fair point. I know this doesn't give you the same thing, but with a named return for it,
you can still avoid typing that and instead type the variable name:

```
func parsePluginOpts(providerOpts workspace.PluginOptions, k workspace.PluginKind) (plug workspace.ProjectPlugin, err error) {
  // ...
  if cond {
    return plug, errors.New("whatever")
  }
```